### ISSUE TITLE
Caching user-defined data

### ISSUE THREAD

**Issue by ntoles** [PrincetonUniversity/Iseult#36]
Currently data that is loaded from disk is cached. If I have a user-defined function that requires a heavy calculation, is its output array being cached, or is it recomputed every time? 
We are thinking of using the user-defined functions to plot 2D temperature maps which would require going over particles. if this has to be recomputed on every time step, this may lag.

**Comment by bcaddy** [comment]
I don't believe that outputs are currently cached. I wouldn't worry about it unless it turns out to be a big issue, my guess is that the matplotlib calls to plot the temperature will take far more time than the numpy calls to compute the temperature.

**Comment by ntoles** [comment]
Plotting 2D arrays is fast. Computing T requires binning and looping over all particles. This will be slow.

**Comment by bcaddy** [comment]
I'm not 100% sure. I think it wouldn't be too challenging but it would require some moderate modifications. Maybe a week to have a version working?
